apps/src/templates/TeachersBeyondHoc.jsx

```diff
@@ -1,22 +1,70 @@
-import React, { Component } from 'react';
+import React, { Component, PropTypes } from 'react';
 import i18n from "@cdo/locale";
 import color from "../util/color";
+import {pegasus} from '@cdo/apps/lib/util/urlHelpers';
+import Responsive from '../responsive';
+import VerticalImageResourceCard from './VerticalImageResourceCard';
+import ResourceCardResponsiveContainer from './studioHomepages/ResourceCardResponsiveContainer';
 
 const styles = {
   heading: {
     color: color.teal,
     width: '100%'
+  },
+  clear: {
+    clear: 'both'
   }
 };
 
 export default class TeachersBeyondHoc extends Component {
+  static propTypes = {
+    isRtl: PropTypes.bool.isRequired,
+    responsive: PropTypes.instanceOf(Responsive).isRequired,
+  };
 
   render() {
+    const { isRtl, responsive } = this.props;
+    const desktop = (responsive.isResponsiveCategoryActive('lg') || responsive.isResponsiveCategoryActive('md'));
+
+    const cards = [
+      {
+        title: i18n.congratsTeacherCodeOrgTitle(),
+        description: i18n.congratsTeacherCodeOrgDesc(),
+        buttonText: i18n.congratsTeacherCodeOrgButton(),
+        link: "/courses?view=teacher",
```

Comment:
I wonder if our React code might need a kind of routes at some point, similar to how Rails provides that?